modules/xfem/src/base/XFEMGeometricCut2D.C

```diff
@@ -102,54 +111,57 @@ bool
 XFEMGeometricCut2D::cutFragmentByGeometry(std::vector<std::vector<Point> > & frag_edges,
                                           std::vector<CutEdge> & cut_edges, Real time)
 {
-  //Use the algorithm described here to determine whether edges are cut by the cut line:
-  //http://stackoverflow.com/questions/563198/how-do-you-detect-where-two-line-segments-intersect
+  // Use the algorithm described here to determine whether edges are cut by the cut line:
+  // http://stackoverflow.com/questions/563198/how-do-you-detect-where-two-line-segments-intersect
 
   bool cut_frag = false;
   Real tol = 1.e-15;
 
   const Real fraction = cutFraction(time);
 
-  unsigned int n_sides = frag_edges.size();
-
-  for (unsigned int i = 0; i < n_sides; ++i)
+  if (fraction > 0.0)
   {
-    Real seg_originx = frag_edges[i][0](0);
```

Comment:
Note that the duplication of code here that appeared above will be addressed by #6505.